### DISCUSSION TITLE
Deployed on my vercel. worked for months but suddenly not working

### DISCUSSION THREAD

**Discussion by phuanggh** [anuraghazra/github-readme-stats#3817] [Q&A] [answered]
### Describe the bug

I deployed it on my vercel server a few months ago to show the stats including my private repos. One day it suddenly stops working and shows "down time due to github API rate limiting"

### Expected behavior

it shows language stats that includes my private repos normally

### Screenshots / Live demo link

<img width="595" alt="image" src="https://github.com/anuraghazra/github-readme-stats/assets/51040483/aed8ba5a-c204-42af-9e1e-8cb292c4dd7a">

**Comment by magicgopher** [comment]
<img width="738" alt="20240618-004013@2x" src="https://github.com/anuraghazra/github-readme-stats/assets/50815770/c57e4ba8-a929-4c5f-86a9-8f27759f2452">

This happens when include_all_commits is set to true, but it works fine when include_all_commits is set to false. There is something wrong with the include_all_commits parameter.

**Comment by phuanggh** [comment]
> This happens when include_all_commits is set to true, but it works fine when include_all_commits is set to false. There is something wrong with the include_all_commits parameter.

I didn't set `include_all_commits` and its default value is false. I tried to explicitly set it to false but it still didn't work

**Comment by magicgopher** [comment]
> > 当include_all_commits设置为true时会出现这种情况，但当include_all_commits设置为false时可以正常工作。include_all_commits参数有问题。
> 
> 我没有设置`include_all_commits`，它的默认值是 false。我尝试将其明确设置为 false，但仍然不起作用

My url is: https://github-readme-stats.vercel.app/api/top-langs/?username=, I found your url is: https://github-readme-stats-bice-iota-93.vercel.app/api/top-langs/?username=, and I previously set include_all_commits=true but it didn't work, then set it to false or remove the include_all_commits parameter.

**Comment by magicgopher** [comment]
> > 当include_all_commits设置为true时会出现这种情况，但当include_all_commits设置为false时可以正常工作。include_all_commits参数有问题。
> 
> 我没有设置`include_all_commits`，它的默认值是 false。我尝试将其明确设置为 false，但仍然不起作用

after I modified the url.
<img width="1035" alt="20240618-184144@2x" src="https://github.com/anuraghazra/github-readme-stats/assets/50815770/3b104e97-3e83-424c-9d1f-7486c73aa706">

**Comment by phuanggh** [comment]
@MagicGopher thanks for answering me. 
The URL you provided is a public one, which does not show the stats of my private repo. 
To show my procate statistics I deployed my own vercel instance so that my URL is different from the public one. It worked for months, but now it shows "downtime due to github API rate limiting" no matter I set include_all_commits=true or include_all_commits=false

**Comment by qwerty541** [comment, marked as answer]
Hey @phuanggh, By the following URL https://github-readme-stats-bice-iota-93.vercel.app/api/status/pat-info you can check that your GitHub API token currently expired. Therefore, you have to generate a new one (without expiration date preferably) and replace with it value of "PAT_1" environment variable on your Vercel deployment. For the future, such error message "Downtime due to GitHub API rate limiting" always shows that something wrong with your GitHub API token.

**Reply by phuanggh** [reply]
Hi @qwerty541 thanks for answering me. I did generate a new GitHub token and replaced it with the value of PAT_1 environment variable after I found it expired, but it's still not working. Do I need to do anything after I replace PAT_1 environment variable?

**Reply by phuanggh** [reply]
I found that I need to redeploy it. Now it works!
Thanks mate!

**Reply by qwerty541** [reply]
Glad to help! Feel free to contact me if you need any further assistance.
